auth/providers/azure/pop_tokenverifier.go

```diff
@@ -136,6 +143,28 @@ func (p *PoPTokenVerifier) ValidatePopToken(token string) (string, error) {
 		return "", errors.Errorf("Invalid token. 'u' claim is missing")
 	}
 
+	// Verify host 'nonce' claim
+	if nonce, ok := claims["nonce"]; ok {
+		if _, ok := nonce.(string); !ok {
+			return "", errors.Errorf("Invalid token. 'nonce' claim should be of string")
+		}
+	} else {
+		return "", errors.Errorf("Invalid token. 'nonce' claim is missing")
+	}
+	// Making sure nonce is not reused
+	if _, ok := nonceMap[claims["nonce"].(string)]; ok {
```

Comment:
Maps aren't thread safe, we need to have all access behind a lock:
https://stackoverflow.com/questions/36167200/how-safe-are-golang-maps-for-concurrent-read-write-operations